cmd/kueue/main.go

```diff
@@ -114,22 +114,37 @@ func main() {
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	if err := utilfeature.DefaultMutableFeatureGate.Set(featureGates); err != nil {
-		setupLog.Error(err, "Unable to set flag gates for known features")
+	options, cfg, err := apply(configFile)
+	if err != nil {
+		setupLog.Error(err, "Unable to load the configuration")
+		os.Exit(1)
+	}
+
+	if featureGates != "" && cfg.FeatureGates != nil {
+		err := errors.New("conflicting feature gates detected")
+		setupLog.Error(err, "Unable to decide which features gate to use due to --feature-gates specified at CLI and featureGates specified in the configuration file")
+		// To avoid future bugs, we should fail to start Kueue if feature gates
+		// are specified for both the CLI and the configuration.
```

Comment:
I think this statement about "avoiding future bugs" is vague, and not visible to the end-users. Also, I think we may want to actually in the future relax this and handle the config map as override. So, I would suggest to drop this comment and update the API field description (as proposed in another comment).

Reply:
updated. PTAL.